### PR TITLE
XW-1722 | Show Global Footer only where needed

### PR DESCRIPTION
**Pull request by rogatty**
## Links

* https://wikia-inc.atlassian.net/browse/XW-1722
* https://github.com/Wikia/mercury/pull/2614

## Description

Add `show-global-footer` class to the `body` element on routes who already display the old Global Nav, on backend and frontend.

Use this class to decide if we should show or hide the footer.

Create `mixins/route-with-body-class-name.js` to make it easier to control body classes from routes.

## Reviewers

@Wikia/x-wing

### CODE REVIEW

File: front/main/app/styles/component/_global-footer.scss [hunk 1]
```diff
@@ -0,0 +1,9 @@
+body {
```
Comment (kamilkoterba) [inline front/main/app/styles/component/_global-footer.scss]: Would these styles work without body nesting?

File: front/main/app/mixins/route-with-body-class-name.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import Ember from 'ember';
+
+const {$, on, Mixin} = Ember;
+
+export default Mixin.create({
+	addBodyClassOnActivate: on('activate', function () {
+		const $body = $('body');
+
+		this.get('bodyClassNames').forEach((className) => {
```
Comment (kvas-damian) [inline front/main/app/mixins/route-with-body-class-name.js]: Can we define default value for `bodyClassNames`?

File: front/main/app/styles/component/_global-footer.scss (hunk 1 above)
Comment (kamilkoterba) [inline front/main/app/styles/component/_global-footer.scss]: Also name of this file may be a bit misleading as those are not global footer styles but just for hide/display

File: front/main/app/styles/component/index.scss [hunk 3]
```diff
@@ -61,6 +61,7 @@
 @import 'potential-member-page-experiment';
 @import 'recirculation-experiment';
 @import 'discussion-reply-editor-entry-point';
+@import 'global-footer';
```
Comment (kamilkoterba) [inline front/main/app/styles/component/index.scss]: The name of this file may be a bit misleading as those are not global footer styles but just for hide/display

File: front/main/app/routes/discussion/guidelines.js [hunk 4]
```diff
@@ -1,29 +1,18 @@
 import Ember from 'ember';
 import DiscussionBaseRoute from './base';
+import RouteWithBodyClassNameMixin from '../../mixins/route-with-body-class-name';
 
-export default DiscussionBaseRoute.extend({
+export default DiscussionBaseRoute.extend(
+	RouteWithBodyClassNameMixin,
+	{
+		bodyClassNames: ['standalone-page'],
```
Comment (kvas-damian) [inline front/main/app/routes/discussion/guidelines.js]: Can you make sure `bodyClassNames` will be merged with value defined in parent, so footer will be visible on guidelines page (`/d/g`)?

**Comment by kamilkoterba** [comment]
Besides comments above LGTM

File: front/main/app/routes/discussion/guidelines.js (hunk 4 above)
Comment (rogatty) [inline front/main/app/routes/discussion/guidelines.js]: It's working fine, Ember magic :sparkles:

Comment (kvas-damian) [inline front/main/app/routes/discussion/guidelines.js]: :+1:

**Pull request merged**